lib/linguist/heuristics.yml

```diff
@@ -94,12 +94,33 @@ disambiguations:
     pattern: '^[ \t]*#(?i)(?:define|endif|endmacro|ifn?def|include|lang|macro)(?:$|\s)'
   - language: BASIC
     pattern: '\A\s*\d'
+  - language: QuickBASIC
+    pattern:
+    - '(#|$)lang:?\s*"?qb"?'
+    - "(?i)'\\$INCLUDE:"
+    - "(?i)^[ ]*CLS[ ]*(''|:|$)"
+    - '(?i)^[ ]*OPTION _EXPLICIT'
+    - '(?i) As _(Byte|Offset|MEM)'
+    - '(?i)^[ ]*_(DISPLAY|DEST|CONSOLE|SOURCE|FREEIMAGE|PALETTECOLOR|PRINTSTRING|LOADFONT|PUTIMAGE)'
+    - '(?i)^[ ]*_(TITLE|PLAYMOD) "'
+    - '(?i)^[ ]*_(LIMIT|SCREEN|DELAY) \.?\d+'
+    - '(?i)\b_(MOUSEBUTTON|NEWIMAGE|KEYDOWN|WIDTH|HEIGHT)\('
+    - '(?i)^[ ]*\$(CONSOLE|CHECKING):'
+    - '(?i)^[ ]*\$(FULLSCREEN|RESIZE|STATIC|DYNAMIC|NOPREFIX|SCREENSHOW|SCREENHIDE|EXEICON)\b'
   - language: VBA
     and:
     - named_pattern: vb-module
     - named_pattern: vba
   - language: Visual Basic 6.0
     named_pattern: vb-module
+  # If no other language is matched, we default to QuickBASIC as long as
+  # there is one upppercase QuickBASIC command.
+  - language: QuickBASIC
+    pattern:
+    - '^[ ]*(CONST|DIM|REDIM|DEFINT|PRINT|DECLARE (SUB|FUNCTION)|FUNCTION|SUB) '
+  # Otherwise, we default to VBA as long as there is a proper Sub or Function definition.
+  - language: VBA
+    pattern: '^[ ]*(Private |Public )?(Sub|Function) '
```

Comment:
Please don't do this... you're defining VBA and QuickBasic multiple times in the same heuristic. This indicates your heuristics have scope for improvement. This is also confusing and will lead to issues in future.

/cc @Alhadis for some regex skills.

Reply:
As far as I know, there is no way to do what I'm trying to do without having more than one entry per language for the `.bas` extension. I assumed that it was fair game since this also occurs for the `.properties` extension and the `.1in` extension. However, I remain open to suggestions if there is a better way.

Reply:
For the second QuickBASIC entry, I could actually merge them together considering that VBA/VB6 use the propercase convention and code that was edited in the VB Editor will never have uppercase capitalization for keywords due case enforcing (as discussed [here](https://github.com/github-linguist/linguist/pull/6355#issuecomment-1493603105)). 

Technically, you could have uppercase keywords if you edit VBA/VB6 code while using another editor and it would be converted seemlessly when added to a VB Project, but I haven't seen any cases of that on GitHub. However, GitHub search is case-insensitive, so it's hard to check.

Reply:
I've merged the 2 QuickBASIC entries and removed the 2nd VBA entry for now. We can re-assess if it's needed after QB is added. Note however that I needed to use 2 consecutive entries for FreeBASIC for the same reason that the Roff Manpage heuristic needs it.